nums/numpy/api.py

```diff
@@ -606,6 +618,47 @@ def average(
     return avg, weights_sum
 
 
+def median(a: BlockArray, axis=None, out=None, keepdims=False) -> BlockArray:
+    """Compute the median of a BlockArray.
+
+    Args:
+        a: A BlockArray.
+
+    Returns:
+        The median value.
+    """
+    if axis is not None:
+        raise NotImplementedError("'axis' is currently not supported.")
+    if out is not None:
+        raise NotImplementedError("'out' is currently not supported.")
+    if keepdims:
+        raise NotImplementedError("'keepdims' is currently not supported.")
+    return _instance().median(a)
+
+
+def top_k(
+    a: BlockArray, k: int, largest=True, sorted=False
+) -> Tuple[BlockArray, BlockArray]:
+    """Find the `k` largest or smallest elements of a BlockArray.
+
+    If there are multiple kth elements that are equal in value, then no guarantees are made as
+    to which ones are included in the top k.
+
+    Args:
+        a: A BlockArray.
+        k: Number of top elements to return.
+        largest: Whether to return largest or smallest elements.
+
+    Returns:
+        A tuple containing two BlockArrays, (`values`, `indices`).
+        values: Values of the top k elements, unsorted.
+        indices: Indices of the top k elements, ordered by their corresponding values.
+    """
+    if sorted:
```

Comment:
If you have an idea on how to support this op with the `sorted` argument, you could note it here as a comment.